### PR TITLE
Name & prop alignment  with Carbon

### PR DESCRIPTION
**Pull request by davidicus**
**Summary**

This PR aligns the names and prop models between Carbon components and our extended version. The goal being to support the same props as the underlining component in addition to our extended props. This PR includes all components except `Table`

**Change List (commits, features, bugs, etc)**

- There are a ton of file changes here, apologies.

**Related Issues**

- Part of IBM/carbon-addons-iot-react#369

**Acceptance Test (how to verify the PR)**

- This would have to be a manual process of checking props between the two repos.

### CODE REVIEW

File: src/styles.scss [hunk 1]
```diff
@@ -175,3 +175,5 @@ $deprecations--message: 'Deprecated code was found, this code will be removed be
 // carbon-addons-iot-react components
 //-------------------------------------
 @import './components/AddCard/add-card';
+@import './components/ProgressIndicator/progress-indicator';
+@import './components/SideNav/side-nav';
```
Comment (tay1orjones) [inline src/styles.scss]: We should add these to the rollup config to be exported/available individually

like 
https://github.com/IBM/carbon-addons-iot-react/pull/432/files#diff-ff6e5f22a9c7e66987b19c0199636480R134

**Comment by davidicus** [comment]
@tay1orjones I think I have it now.

File: rollup.config.js [hunk 2]
```diff
@@ -132,6 +132,11 @@ export default {
           dest: 'lib/scss/components',
         },
         { src: ['src/components/AddCard/_add-card.scss'], dest: 'lib/scss/components/AddCard' },
+        {
+          src: ['src/components/ProgressIndicator/progress-indicator.scss'],
```
Comment (tay1orjones) [inline rollup.config.js]: I think this needs an underscore

**Pull request merged**